### PR TITLE
Upgrade to Apache Beam, version 0.6.0

### PR DESCRIPTION
**Pull request by jasonkuster**
Lots of manual changes -- hopefully some which will make this process easier in the future.

Verified via mvn clean install, then generating, and then running WordCount in both examples and examples-java8 and StarterPipeline from starter.

R: @davorbonaci @aaltay

### CODE REVIEW

**Comment by jasonkuster** [comment]
Note: As-is, will not pass -- need to update container image to 2.0.0-beta3 (omitting -20170202); 20170202 container is missing changes from Beam 0.6.0.

File: maven-archetypes/pom.xml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
```
Comment (davorbonaci) [inline maven-archetypes/pom.xml]: Google header goes here.

File: maven-archetypes/starter/pom.xml [hunk 2]
```diff
@@ -41,5 +42,44 @@
         <version>2.4</version>
       </extension>
     </extensions>
+
+    <pluginManagement>
+      <plugins>
+        <plugin>
+          <artifactId>maven-archetype-plugin</artifactId>
```
Comment (davorbonaci) [inline maven-archetypes/starter/pom.xml]: Note to self: I wonder why this is not in the parent?

File: maven-archetypes/pom.xml (hunk 1 above)
Comment (jasonkuster) [inline maven-archetypes/pom.xml]: Oops, fixed.

**Pull request merged**